### PR TITLE
 consume leading dots when converting from carbon

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

File: route/schemas.go [hunk 1]
```diff
@@ -63,6 +65,15 @@ func parseMetric(buf []byte, schemas persister.WhisperSchemas, orgId int) (*sche
...
 
+	// skip over all leading dots, like graphite does
+	// if the key is nothing but dots, it's invalid
+	for elements[0][0] == '.' {
+		if len(elements[0]) == 1 {
+			return nil, errInvalidKey
+		}
+		elements[0] = elements[0][1:]
+	}
+
```
Comment (DanCech) [inline route/schemas.go]: https://golang.org/pkg/strings/#TrimLeft

**Comment by robert-milan** [comment]
Closing in favor of #375

**Comment by Dieterbe** [comment]
That PR doesn't seem to do enough or what this PR is about. Remember that most of this project is about ingesting, manipulating and emitting carbon streams
 Schema lib is only used for select output plugins such as grafanaNet output

**Comment by robert-milan** [comment]
I'll re-open this until I update M20 to use `EatDots` and ensure it is enforced in the ingest path here.

**Comment by npazosmendez** [comment]
I'm doing some housekeeping in this repo, I'll close this PR as it seems stale. Feel free to reopen if you still think this is worth it.

**Pull request closed without merging**